tests/kafkatest/sanity_checks/test_performance_services.py

```diff
@@ -38,15 +38,9 @@ def setUp(self):
             self.zk.start()
 
     @cluster(num_nodes=5)
-    # We are keeping 0.8.2 here so that we don't inadvertently break support for it. Since this is just a sanity check,
-    # the overhead should be manageable.
-    @parametrize(version=str(LATEST_0_8_2), new_consumer=False)
-    @parametrize(version=str(LATEST_0_9), new_consumer=False)
-    @parametrize(version=str(LATEST_0_9))
-    @parametrize(version=str(LATEST_1_1), new_consumer=False)
-    @cluster(num_nodes=5)
+    @parametrize(version=str(LATEST_2_1), new_consumer=False)
     @matrix(version=[str(DEV_BRANCH)], metadata_quorum=quorum.all)
```

Comment:
Could you please use `@matrix(version=[str(LATEST_2_1), str(DEV_BRANCH)], metadata_quorum=quorum.all)` instead?

Reply:
I tried to update it, but it looks like `LATEST_2_1` can't pass for this case:

```
================================================================================
SESSION REPORT (ALL TESTS)
ducktape version: 0.12.0
session_id:       2024-11-09--005
run time:         3 minutes 6.672 seconds
tests run:        6
passed:           3
flaky:            0
failed:           3
ignored:          0
================================================================================
test_id:    kafkatest.sanity_checks.test_performance_services.PerformanceServiceTest.test_version.version=2.1.1.metadata_quorum=COMBINED_KRAFT
status:     FAIL
run time:   31.068 seconds


    TypeError("'NoneType' object is not subscriptable")
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/ducktape/tests/runner_client.py", line 351, in _do_run
    data = self.run_test()
  File "/usr/local/lib/python3.7/dist-packages/ducktape/tests/runner_client.py", line 411, in run_test
    return self.test_context.function(self.test)
  File "/usr/local/lib/python3.7/dist-packages/ducktape/mark/_mark.py", line 438, in wrapper
    return functools.partial(f, *args, **kwargs)(*w_args, **w_kwargs)
  File "/opt/kafka-dev/tests/kafkatest/sanity_checks/test_performance_services.py", line 80, in test_version
    consumer_perf_data = compute_aggregate_throughput(self.consumer_perf)
  File "/opt/kafka-dev/tests/kafkatest/services/performance/performance.py", line 69, in compute_aggregate_throughput
    aggregate_rate = sum([r['records_per_sec'] for r in perf.results])
  File "/opt/kafka-dev/tests/kafkatest/services/performance/performance.py", line 69, in <listcomp>
    aggregate_rate = sum([r['records_per_sec'] for r in perf.results])
TypeError: 'NoneType' object is not subscriptable

--------------------------------------------------------------------------------
test_id:    kafkatest.sanity_checks.test_performance_services.PerformanceServiceTest.test_version.version=2.1.1.metadata_quorum=ISOLATED_KRAFT
status:     FAIL
run time:   39.534 seconds


    TypeError("'NoneType' object is not subscriptable")
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/ducktape/tests/runner_client.py", line 351, in _do_run
    data = self.run_test()
  File "/usr/local/lib/python3.7/dist-packages/ducktape/tests/runner_client.py", line 411, in run_test
    return self.test_context.function(self.test)
  File "/usr/local/lib/python3.7/dist-packages/ducktape/mark/_mark.py", line 438, in wrapper
    return functools.partial(f, *args, **kwargs)(*w_args, **w_kwargs)
  File "/opt/kafka-dev/tests/kafkatest/sanity_checks/test_performance_services.py", line 80, in test_version
    consumer_perf_data = compute_aggregate_throughput(self.consumer_perf)
  File "/opt/kafka-dev/tests/kafkatest/services/performance/performance.py", line 69, in compute_aggregate_throughput
    aggregate_rate = sum([r['records_per_sec'] for r in perf.results])
  File "/opt/kafka-dev/tests/kafkatest/services/performance/performance.py", line 69, in <listcomp>
    aggregate_rate = sum([r['records_per_sec'] for r in perf.results])
TypeError: 'NoneType' object is not subscriptable

--------------------------------------------------------------------------------
test_id:    kafkatest.sanity_checks.test_performance_services.PerformanceServiceTest.test_version.version=2.1.1.metadata_quorum=ZK
status:     FAIL
run time:   34.182 seconds


    TypeError("'NoneType' object is not subscriptable")
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/ducktape/tests/runner_client.py", line 351, in _do_run
    data = self.run_test()
  File "/usr/local/lib/python3.7/dist-packages/ducktape/tests/runner_client.py", line 411, in run_test
    return self.test_context.function(self.test)
  File "/usr/local/lib/python3.7/dist-packages/ducktape/mark/_mark.py", line 438, in wrapper
    return functools.partial(f, *args, **kwargs)(*w_args, **w_kwargs)
  File "/opt/kafka-dev/tests/kafkatest/sanity_checks/test_performance_services.py", line 80, in test_version
    consumer_perf_data = compute_aggregate_throughput(self.consumer_perf)
  File "/opt/kafka-dev/tests/kafkatest/services/performance/performance.py", line 69, in compute_aggregate_throughput
    aggregate_rate = sum([r['records_per_sec'] for r in perf.results])
  File "/opt/kafka-dev/tests/kafkatest/services/performance/performance.py", line 69, in <listcomp>
    aggregate_rate = sum([r['records_per_sec'] for r in perf.results])
TypeError: 'NoneType' object is not subscriptable

--------------------------------------------------------------------------------
test_id:    kafkatest.sanity_checks.test_performance_services.PerformanceServiceTest.test_version.version=dev.metadata_quorum=COMBINED_KRAFT
status:     PASS
run time:   24.350 seconds
{"producer_performance": {"records_per_sec": 35971.2, "mb_per_sec": 3.43}, "end_to_end_latency": {"latency_50th_ms": 0.0, "latency_99th_ms": 1.0, "latency_999th_ms": 3.0}, "consumer_performance": {"records_per_sec": 30091.954, "mb_per_sec": 2.7676}}
--------------------------------------------------------------------------------
test_id:    kafkatest.sanity_checks.test_performance_services.PerformanceServiceTest.test_version.version=dev.metadata_quorum=ISOLATED_KRAFT
status:     PASS
run time:   29.257 seconds
{"producer_performance": {"records_per_sec": 33898.3, "mb_per_sec": 3.23}, "end_to_end_latency": {"latency_50th_ms": 0.0, "latency_99th_ms": 1.0, "latency_999th_ms": 3.0}, "consumer_performance": {"records_per_sec": 31829.7872, "mb_per_sec": 2.9274}}
--------------------------------------------------------------------------------
test_id:    kafkatest.sanity_checks.test_performance_services.PerformanceServiceTest.test_version.version=dev.metadata_quorum=ZK
status:     PASS
run time:   27.490 seconds
{"producer_performance": {"records_per_sec": 32362.5, "mb_per_sec": 3.09}, "end_to_end_latency": {"latency_50th_ms": 0.0, "latency_99th_ms": 1.0, "latency_999th_ms": 4.0}, "consumer_performance": {"records_per_sec": 30265.896, "mb_per_sec": 2.7836}}
--------------------------------------------------------------------------------
```

Reply:
If we only apply this change on trunk branch, we also get same error. The reason is that `last` is `None`.

```
        last = None
        for line in node.account.ssh_capture(cmd):
            last = line

        # Parse and save the last line's information
        self.results[idx-1] = self.parse_results(last, node.version)
```

Reply:
@chia7712, I found the root cause. In 2.1.1, we still use broker-list, not bootstrap-server.

https://github.com/apache/kafka/blob/21234bee31165527859b46ea48c46b76532f7a37/core/src/main/scala/kafka/tools/ConsumerPerformance.scala#L209-L212

Reply:
New result:

```
================================================================================
SESSION REPORT (ALL TESTS)
ducktape version: 0.12.0
session_id:       2024-11-09--008
run time:         2 minutes 45.572 seconds
tests run:        6
passed:           6
flaky:            0
failed:           0
ignored:          0
================================================================================
```